src/khepri_fun.erl

```diff
@@ -445,22 +453,57 @@ handle_validation_error(
     {bad_type, {needed, NeededType}, {actual, any}}}},
   Error) ->
     VarInfo = {var_info, Var, [{type, NeededType}]},
-    Comment = {'%', VarInfo},
+    Comments = [{'%', VarInfo}],
     Location = {before, FailingInstruction},
-    add_comment_and_retry(Asm, Error, FailingFun, Location, Comment);
+    add_comments_and_retry(Asm, Error, FailingFun, Location, Comments);
 handle_validation_error(Asm, _ValidationFailure, Error) ->
     throw({compilation_failure, Error, Asm}).
 
-add_comment_and_retry(
-  Asm, Error, FailingFun, Location, Comment) ->
+find_comments_for_call(
+  [#function{name = Name, arity = Arity, code = Instructions} | _Rest],
+  Name, Arity, CallIndex) ->
```

Comment:
Could you please add a comment which explains the logic of `find_comments_for_call()` and `find_comments_in_branch()`? Perhaps with a short example of the assembly.

I find it handy to have comments like this, so that we don't have to go to the commit message to understand what's going on.

Reply:
I ended up refactoring out `find_comments_for_call` since it was really just duplicating `lookup_function1/3` and then calling out to `find_comments_in_branch/2`. It seems more composable to keep that logic in `lookup_function1/3` (but maybe we should rename that function now?)

Reply:
Good idea!